docs/rules/spec-strict-refs.md

```diff
@@ -0,0 +1,99 @@
+# spec-strict-refs
+
+Restricts the usage of the `$ref` keyword.
+
+|OAS|Compatibility|
+|---|---|
+|2.0|✅|
+|3.0|✅|
+|3.1|✅|
+
+## API design principles
+
+This rule is for spec correctness.
+
+Allows to use the `$ref` keyword exclusively for references to elements that may be inside the component section
```

Comment:
Let's also explain that this is useful if you have other tools in your API workflow that require tight compliance with the spec.